src/constructs/lambda/lambda.ts

```diff
@@ -71,3 +77,21 @@ export class GuLambdaFunction extends Function {
     AppIdentity.taggedConstruct(props, this);
   }
 }
+
+/**
+ * Creates a [[`GuLambdaFunction`]] with the latest Java runtime.
+ */
+export class GuJvmLambdaFunction extends GuLambdaFunction {
+  constructor(scope: GuStack, id: string, props: GuRuntimeSpecificFunctionProps) {
+    super(scope, id, { ...props, runtime: latestJava });
+  }
+}
+
+/**
+ * Creates a [[`GuLambdaFunction`]] with the latest Node runtime.
+ */
+export class GuNodeLambdaFunction extends GuLambdaFunction {
```

Comment:
AWS CDK has native [support for NodeJsFunctions](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-lambda-nodejs-readme.html). A brief look at the docs suggests it has support for TS, which massively simplifies the build step!

Do we want to make use of that construct?

I think this is an example of "don't build your own constructs" that [this CDK Day talk](https://www.youtube.com/watch?v=1gxArqL_Eg4) describes.

Reply:
I think I agree with Akash on the sentiment that we may want to consider creating fewer GuConstructs if we can get away with using AWS ones. I haven't watched the talk yet though, and also agree with the logic that the users should decide whether the constructs are fit for purpose or not, so not sure how strict we need to be with that.

So, I guess I wonder if the GuNodeLambdaFunction could just be a wrapper for the AWS CDK native construct, meaning we don't have to maintain it as much?

Reply:
As the AWS construct seems to take a very different approach to building/deploying Node lambda code, I think it'd be worth understanding that in more detail (via a spike) before adding our own implementation of a Node lambda.